urlc58f2e0b85_xoriole_CEAC_tgz-pJ8-com_sdm_phr_cpabe_CpabeTestJ8/src/com/sdm/phr/cpabe/Common.java

```diff
@@ -0,0 +1,217 @@
+package com.sdm.phr.cpabe;
+
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+
+public class Common {
+
+    /* read byte[] from inputfile */
+    public static byte[] suckFile(String inputfile) throws IOException {
+        InputStream is = new FileInputStream(inputfile);
+        int size = is.available();
+        byte[] content = new byte[size];
+
+        is.read(content);
+
+        is.close();
+        return content;
+    }
+
+    /* write byte[] into outputfile */
+    public static void spitFile(String outputfile, byte[] b) throws IOException {
+        OutputStream os = new FileOutputStream(outputfile);
+        os.write(b);
+        os.close();
+    }
+
+    public static void writeCpabeFile(String encfile,
+            byte[] cphBuf, byte[] aesBuf) throws IOException {
+        int i;
+        OutputStream os = new FileOutputStream(encfile);
+
+        /* write aes_buf */
+        for (i = 3; i >= 0; i--) {
+            os.write(((aesBuf.length & (0xff << 8 * i)) >> 8 * i));
+        }
+        os.write(aesBuf);
+
+        /* write cph_buf */
+        for (i = 3; i >= 0; i--) {
+            os.write(((cphBuf.length & (0xff << 8 * i)) >> 8 * i));
+        }
+        os.write(cphBuf);
+
+        os.close();
+
+    }
+
+    /**
+     * 
+     * @param cphBuf
+     * @param aesBuf
+     * @return
+     * @throws IOException 
+     */
+    public static byte[] writeCpabeFile(byte[] cphBuf, byte[] aesBuf) 
+            throws IOException {
+        int i;
+        ByteArrayOutputStream os = new ByteArrayOutputStream();
+
+        /* write aes_buf */
+        for (i = 3; i >= 0; i--) {
+            os.write(((aesBuf.length & (0xff << 8 * i)) >> 8 * i));
+        }
+        os.write(aesBuf);
+
+        /* write cph_buf */
+        for (i = 3; i >= 0; i--) {
+            os.write(((cphBuf.length & (0xff << 8 * i)) >> 8 * i));
+        }
+        os.write(cphBuf);
+
+        return os.toByteArray();
+    }
+
+    public static byte[][] readCpabeFile(String encfile) throws IOException {
+        int i, len;
+        InputStream is = new FileInputStream(encfile);
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code contains a resource that might not be closed properly. This can cause a resource leak that slows down or crashes your system.

**Fix**
Consider closing the following resource: *is*. The resource is referenced at lines 89, 93, 98, 102. The resource is closed at line 104. There are other execution paths that do not contain closure statements, for example, when InputStream.read throws an exception. Ensure *is* is closed in a try-finally block or declared in a try-with-resources block.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).